pkg/utils/db.go

```diff
@@ -424,3 +427,44 @@ func GetGTID(db *sql.DB) (string, error) {
 	val, err := GetGlobalVariable(db, "GTID_MODE")
 	return val, err
 }
+
+// ToTiDBVersion convert string to TiDBVersion
+// version format: 5.7.25-TiDB-v3.0.7
+func ToTiDBVersion(v string) (TiDBVersion, error) {
```

Comment:
TiDB-Lightning also has its implementation, https://github.com/pingcap/tidb-lightning/blob/e4de23b08259514ca20797696b6cd021a6a6de1a/lightning/backend/importer.go#L256

you may re-check with it.